File: frontend/src/types/workflow.ts
export interface WorkflowProfile {
  id: string;
  display_name: string;
  description: string;
  aspect_ratio: string;
  padding_mode: string;
  allow_intro_outro: boolean;
  enable_subtitles: boolean;
  subtitle_preset: string;
  enable_thumbnails: boolean;
  export_preset: string;
}

export interface AIContentSuggestions {
  titles: string[];
  description: string;
  hashtags: string[];
  chapters: string[];
}

File: backend/app/models/workflow.py
from typing import Optional
from pydantic import BaseModel, Field

class WorkflowProfile(BaseModel):
    id: str
    display_name: str
    description: str
    aspect_ratio: str = Field(default="original")
    padding_mode: str = Field(default="black_bars")
    allow_intro_outro: bool = Field(default=True)
    enable_subtitles: bool = Field(default=False)
    subtitle_preset: str = Field(default="tiktok")
    enable_thumbnails: bool = Field(default=True)
    export_preset: str = Field(default="high_quality")
